src/os/ui/layer/abstractlayerui.js

```diff
@@ -159,6 +159,11 @@ os.ui.layer.AbstractLayerUICtrl.prototype.getLayerNodes = function() {
   var nodes = [];
 
   var items = this.scope['items'] || [];
+  for (var i = 0; i < items.length; i++) {
```

Comment:
This is too specific for an abstract class. I would recommend making a new layer UI that extends `os.ui.layer.ImageLayerUI` (something like `plugin.file.kml.ui.KMLImageNodeUI`) and provides its own `getLayerNodes` function. When the image layer is created by the KML parser, use `setLayerUI` to override the default UI for the layer.